frontend/app.py

```diff
@@ -1,22 +1,20 @@
-# %%
 import requests
 import streamlit as st
 import streamlit.components.v1 as components
 
 
 def get_data(query: str, endpoint: str) -> dict:
-    headers = {
-        "Content-Type": "application/json",
-    }
 
-    data = '{"data":[{"text": "' + query + '"}]}'
+    content = requests.post(
+        endpoint,
+        headers={
+            "Content-Type": "application/json",
+        },
+        data=f'{{"data":[{{"text": "{query}"}}]}}',
+    ).json()
```

Comment:
Why this refactoring? What is the rationale? (Just curious, looks good for me).

Reply:
The previous code introduced variables that were consumed immediately and not anywhere else. For example,

```python
def get_data(query: str, endpoint: str) -> dict:

   # This can be specified directly in `.post` without making
   # a one time variable
    headers = {
        "Content-Type": "application/json",
    }
 
   # Same as above, since it is used only in one place
   # ... might as well directly feed it where needed
    data = '{"data":[{"text": "' + query + '"}]}'
    
    # Get the response in one line, then the content in another?
    # There is no expected interaction with anything else but `.json`
    # from the response - this can be a one liner
    response = requests.post(endpoint, headers=headers, data=data)
    content = response.json()
    
    # Extract contents here? This could be changed to just response.json()['data']['docs']
    matches = content["data"]["docs"]

    # Making a variable `matches` just to immediately return it 
    # We can instead do something above like `return response.json()['data']['docs']`
    return matches
 ```
 
 However writing `.json['data']['docs']` looks "ugly" to me for some reason, so the new refactor could directly return data without storing it in `content` as well, but I did not go with that because the `return content['data']['docs']` does not make you think what it is doing, but `.json()['data']['docs']` does.

Reply:
```python
# This can be specified directly in `.post` without making
# a one time variable
headers = {
  "Content-Type": "application/json",
}
```
Agreed, given that the header is constant.

```python
# Same as above, since it is used only in one place
# ... might as well directly feed it where needed
data = '{"data":[{"text": "' + query + '"}]}'
```
This is something that is meant to change during development (improving the frontend, we're gonna query more data and make this request more complex) so it would be nice to have some space and separate this from the rest of the code.

```python
# Get the response in one line, then the content in another?
# There is no expected interaction with anything else but `.json`
# from the response - this can be a one liner
response = requests.post(endpoint, headers=headers, data=data)
content = response.json()
```
Here the idea is "one line does one thing" (it applies also to the above, in a certain sense). If I call 2 functions, I like to see those on 2 lines. Might be more readable sometimes, even thought I agree that in this case we can safely merge the 2 lines.

Reply:
> This is something that is meant to change during development (improving the frontend, we're gonna query more data and make this request more complex) so it would be nice to have some space and separate this from the rest of the code.

Noted. I'll revert these changes.

> Here the idea is "one line does one thing" (it applies also to the above, in a certain sense). If I call 2 functions, I like to see those on 2 lines. Might be more readable sometimes, even though I agree that in this case, we can safely merge the 2 lines.

Revert is needed here.